### PR TITLE
Removing IVT where possible Attempt 2

### PR DESCRIPTION
**Pull request by Zombach**
https://github.com/dotnet/aspire/issues/3949

Made some minor edits and also cleaned up IVT if possible.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3982)

### CODE REVIEW

**Comment by Zombach** [comment]
@eerhardt I hope it will be better this way

**Comment by davidfowl** [comment]
IVT is fine for tests. We don't want many of these APIs to be public. Each of these projects should have their own test project instead of all using the same test project.

@eerhardt we shouldn't be removing IVT to our tests, just between product code.

File: src/Aspire.Hosting.MySql/PhpMyAdminConfigWriterHook.cs [hunk 1]
```diff
@@ -6,8 +6,10 @@
 
 namespace Aspire.Hosting.MySql;
 
-internal sealed class PhpMyAdminConfigWriterHook : IDistributedApplicationLifecycleHook
+/// <inheritdoc/>
+public class PhpMyAdminConfigWriterHook : IDistributedApplicationLifecycleHook
```
Comment (eerhardt) [inline src/Aspire.Hosting.MySql/PhpMyAdminConfigWriterHook.cs:10]: We shouldn't be making things public for this.

https://github.com/dotnet/aspire/pull/3839#discussion_r1579781437

>> Is it worth creating a separate PR to fix this?

> If it is simple, like the case here where it is just trying to reuse constant strings, sure.

**Comment by eerhardt** [comment]
>  we shouldn't be removing IVT to our tests

We don't need to use IVT just to share a couple constant strings, like the docker image name and tag. Many of these tests just need to share the constant strings, and we can just compile them into the tests, like we do elsewhere.

For more complicated things we can leave IVT.

File: tests/Aspire.Hosting.Tests/Redis/AddRedisTests.cs [hunk 2]
```diff
@@ -256,10 +256,16 @@ public void WithDataVolumeAddsPersistenceAnnotation()
...
-        Assert.Equal(TimeSpan.FromSeconds(60), persistenceAnnotation.Interval);
-        Assert.Equal(1, persistenceAnnotation.KeysChangedThreshold);
+        var args = new List<object>();
+        foreach (var argsAnnotation in argsCallbacks)
+        {
+            Assert.NotNull(argsAnnotation.Callback);
+            argsAnnotation.Callback(new CommandLineArgsCallbackContext(args));
+        }
+
+        Assert.Equal("--save 60 1".Split(" "), args);
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Tests/Redis/AddRedisTests.cs:268]: I like testing like this. I think we should keep this change, it actually tests more like a customer. Call the API, and observe the change in behavior.

File: src/Aspire.Hosting.MySql/PhpMyAdminConfigWriterHook.cs (hunk 1 above)
Comment (Zombach) [reply src/Aspire.Hosting.MySql/PhpMyAdminConfigWriterHook.cs]: Yes OK. I canceled it

File: src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs [hunk 3]
```diff
@@ -12,6 +12,7 @@ namespace Aspire.Hosting;
 /// </summary>
 public static class MongoDBBuilderExtensions
 {
+    private const string MongoDBContainerDataDirectory = "/var/mongodb/data";
```
Comment (eerhardt) [inline src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs:15]: Please revert the changes to this file.

1. We shouldn't be making unnecessary/unrelated changes in a PR. Keep each PR to a single change.
2. This isn't correct. From https://stackoverflow.com/questions/35400740/how-to-set-docker-mongo-data-volume it needs to stay `/data/db`.

Have you tested if the data volumes still work?

Comment (Zombach) [reply src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs]: Ready. Canceled.
Sorry. This is because I want to do a little more. I fall for this all the time.
Although I understand perfectly well. What one PR needs to do to solve one problem.

Comment (Zombach) [reply src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs]: > Have you tested if the data volumes still work?

Honestly, no. I just saw that the tests were working. I didn’t run it separately to check data volumes

**Review by eerhardt — APPROVED** [review]

**Comment by eerhardt** [comment]
Thanks for the contribution!
